File: backend/src/main/java/com/example/backend/store/entity/Store.java
package com.example.backend.store.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Entity
@Table(name = "store")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Store {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "store_name", length = 64)
    private String storeName;

    @Column(name = "branch_name", length = 12)
    private String branchName;

    @Column(name = "biz_category_code", length = 6, columnDefinition = "CHAR(6)")
    private String bizCategoryCode;

    @Column(name = "dong_code", nullable = false)
    private Integer dongCode;

    @Column(name = "road_address", length = 32)
    private String roadAddress;

    @Column(name = "lat", precision = 15, scale = 12, nullable = false)
    private BigDecimal lat;

    @Column(name = "lng", precision = 15, scale = 12, nullable = false)
    private BigDecimal lng;
}

File: original/features/stores/types/store.ts
// types/store.ts
export interface Store {
  id: number;
  storeName: string;
  branchName: string;
  bizCategoryCode: string;
  dongCode: number;
  roadAddress: string;
  lat: number;
  lng: number;
  hidden?: boolean;
  // 표시용 필드 (계산된 값)
  displayName?: string;    // storeName + branchName
  categoryName?: string;   // bizCategoryCode → 한글 변환

}
